target_chains/aptos/contracts/sources/pyth.move

```diff
@@ -163,20 +171,113 @@ module pyth::pyth {
+        let message_cur = cursor::init(message);
+        let message_type = deserialize::deserialize_u8(&mut message_cur);
+
+        assert!(message_type == 0, error::invalid_accumulator_message()); // PriceFeedMessage variant
+        let price_identifier = price_identifier::from_byte_vec(deserialize::deserialize_vector(&mut message_cur, 32));
+        let price = deserialize::deserialize_i64(&mut message_cur);
+        let conf = deserialize::deserialize_u64(&mut message_cur);
+        let expo = deserialize::deserialize_i32(&mut message_cur);
+        let publish_time = deserialize::deserialize_u64(&mut message_cur);
+        let _prev_publish_time = deserialize::deserialize_i64(&mut message_cur);
+        let ema_price = deserialize::deserialize_i64(&mut message_cur);
+        let ema_conf = deserialize::deserialize_u64(&mut message_cur);
+        let price_info = price_info::new(
+            timestamp::now_seconds(), // not used anywhere kept for backward compatibility
+            timestamp::now_seconds(),
+            price_feed::new(
+                price_identifier,
+                pyth::price::new(price, conf, expo, publish_time),
+                pyth::price::new(ema_price, ema_conf, expo, publish_time),
+            )
+        );
+        cursor::rest(message_cur);
+        price_info
+    }
 
-        // Deserialize the batch price attestation
-        update_cache(batch_price_attestation::destroy(batch_price_attestation::deserialize(vaa::destroy(vaa))));
+    fun parse_accumulator_updates(cursor: &mut Cursor<u8>, merkle_root: &vector<u8>): vector<PriceInfo> {
+        let update_size = deserialize::deserialize_u8(cursor);
+        let updates: vector<PriceInfo> = vector[];
+        while (update_size > 0) {
+            let message_size = deserialize::deserialize_u16(cursor);
+            let message = deserialize::deserialize_vector(cursor, message_size);
+            let update = parse_accumulator_update_message(message);
+            vector::push_back(&mut updates, update);
+            let path_size = deserialize::deserialize_u8(cursor);
+            let merkle_path: vector<keccak160::Hash> = vector[];
+            while (path_size > 0) {
+                let hash = deserialize::deserialize_vector(cursor, keccak160::get_hash_length());
+                vector::push_back(&mut merkle_path, keccak160::new(hash));
+                path_size = path_size - 1;
+            };
+            assert!(merkle::check(&merkle_path, merkle_root, message), error::invalid_proof());
+            update_size = update_size - 1;
+        };
+        updates
+    }
+
+
+    fun parse_accumulator(cursor: &mut Cursor<u8>): vector<PriceInfo> {
+        let major = deserialize::deserialize_u8(cursor);
+        assert!(major == 1, error::invalid_accumulator_payload());
+        let minor = deserialize::deserialize_u8(cursor);
+        assert!(minor == 0, error::invalid_accumulator_payload());
```

Comment:
I think this check shouldn't be here -- I believe a bump of the minor version corresponds to a forward-compatible message change (like extending the length of a part of the payload) where this code should be able to deserialize it.